### PR TITLE
UI refactor

### LINKED ISSUE #457: UI REFACTOR
# **Description**
**Task:** Refactor and clean up UI
**Feature:** #410 

Upon the request of design committee, the current implementation of inventory display is enhanced and refactored so that when new items are introduced to the game, they are not required to update in many places. Moreover, refactor so that the item are displayed only when they are collected. 
 
## **Example (optional)**
when a new game is started the bottom inventory bar is empty and as we collect more `usable items` they are added and displayed on the screen.

# **Dependencies**
- [x] Player Factory needs to be refactored #424 

# **Milestones**
List of steps that need to be completed for this task.
- [x] Refactor Key binding (Oct 10th)
- [x] Refactor the display (Oct 11th)
- [x] Testing (Oct 13th)
- [x] Update Documentation (Oct 14th)

 
**Completion Deadline:** OCT 16th

# **Documentation**
- [Main description of feature](https://github.com/UQcsse3200/2024-studio-1/wiki/Character's-Inventory-Display)
 
# **Member**
- Manya @manya-k

### PR DESCRIPTION
**Pull request by manya-k**
# Description

Upon request, the current implementation of inventory display is enhanced and refactored so that when new items are introduced to the game, they are not required to update in many places. Moreover, refactor so that the item are displayed only when they are collected.

closes #457 

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Manual Testing

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: source/core/src/main/com/csse3200/game/components/player/PlayerActions.java [hunk 1]
```diff
@@ -34,13 +34,7 @@ public void create() {
         entity.getEvents().addListener("walkStop", this::stopWalking);
         entity.getEvents().addListener("attackMelee", this::attack);
         entity.getEvents().addListener("shoot", this::shoot);
-        entity.getEvents().addListener("use1", () -> use(new MedKit()));
```
Comment (vsinh14) [inline source/core/src/main/com/csse3200/game/components/player/PlayerActions.java:37]: How does one button use all the items?

File: source/core/src/main/com/csse3200/game/components/player/PlayerActions.java [hunk 2]
```diff
@@ -174,20 +168,22 @@ public Vector2 getWalkDirection() {
...
             //Otherwise the reroll item cannot be used
         }
     }
 
-    private void use(UsableItem item) {
+    private void use(int itemNum) {
         if (!dead) {
-            for (Collectible collectedItem : inventoryComponent.getItems()) {
-                if (collectedItem.getClass() == item.getClass()) {
+            System.out.println(itemNum + "being used");
```
Comment (vsinh14) [inline source/core/src/main/com/csse3200/game/components/player/PlayerActions.java:179]: use loggers

**Review by vsinh14 — COMMENTED** [review]
Everything else looks good will test once requested changes are made

File: source/core/src/main/com/csse3200/game/components/player/PlayerActions.java (hunk 1 above)
Comment (manya-k) [reply source/core/src/main/com/csse3200/game/components/player/PlayerActions.java:37]: the use method takes an integer and uses the inventory stored at that index in player's usable inventory array.

File: source/core/src/main/com/csse3200/game/components/player/PlayerActions.java (hunk 2 above)
Comment (manya-k) [reply source/core/src/main/com/csse3200/game/components/player/PlayerActions.java]: fixed

**Pull request merged**